### ISSUE TITLE
Automatically handle HEAD

### ISSUE THREAD

**Issue by jonas-schievink** [1aim/from-request#4]
HEAD is equivalent to GET, except that the response body is omitted. We could automatically route HEAD requests to a matching GET route, unless the route explicitly defines a `#[head]` route. However, this gets complicated in the presence of overlapping routes. Alternatively we could just remove `#[head]` altogether, since changing its behavior (but not `GET`s) seems very unusual.

**Comment by meh** [comment]
What about CORS support? That would require also responding to an `OPTIONS` request for the same resource.

If you need help with that I implemented a couple libraries in various languages that do that.

**Comment by jonas-schievink** [comment]
Yes, I briefly thought about OPTIONS as well, but didn't find a good solution yet. If you could link to your libraries that'd be great.

**Comment by meh** [comment]
https://github.com/meh/urna/blob/master/lib/urna/backend.ex#L28
https://github.com/meh/urna/blob/master/lib/urna.ex#L61

**Comment by jonas-schievink** [comment]
HEAD routes are now added to GET routes automatically, unless the user has defined one manually.
